guide/popular-topics/audit-logs.md

```diff
@@ -1,145 +1,95 @@
-	}
+const fetchedLogs = await guild.fetchAuditLogs({
+	type: AuditLogEvent.InviteCreate,
+	limit: 1,
 });
+
+const firstEntry = fetchedLogs.entries.first();
 ```
 
-With this, you now have a very simple logger telling you who deleted a message authored by another person.
+This will return the first entry where an invite was created. We used `limit: 1` here to specify only one entry.
 
-## Who kicked a user?
+## Receiving Audit Logs
 
-Similar to the `messageDelete` case, let's look at the `guildMemberRemove` event.
+Audit logs may be received via the gateway event `guildAuditLogEntryCreate`. This is the best way to receive audit logs if you are monitoring them. As soon as a message is deleted, or an invite is created, or an emoji is created, you can listen to this event to check information about said occurrence. A common use case is to find out _who_ did something that generated an audit log event.
 
-```js
-client.on(Events.GuildMemberRemove, member => {
-	console.log(`${member.user.tag} left the guild... but was it of their own free will?`);
-});
-```
+### Who deleted a message?
 
-The same as before: set the fetch limit to 1 and accept only the `MemberKick` type.
+One of the most common use cases for audit logs would be understanding who deleted any given message in a Discord server. If another user deleted another user's message, you can find out who did that as soon as you receive the event.
 
-```js {2-7,9-10,12-14,16-22}
-client.on(Events.GuildMemberRemove, async member => {
-	const fetchedLogs = await member.guild.fetchAuditLogs({
-		limit: 1,
-		type: AuditLogEvent.MemberKick,
-	});
-	// Since there's only 1 audit log entry in this collection, grab the first one
-	const kickLog = fetchedLogs.entries.first();
+```js
+const { AuditLogEvent, Events } = require('discord.js');
 
-	// Perform a coherence check to make sure that there's *something*
-	if (!kickLog) return console.log(`${member.user.tag} left the guild, most likely of their own will.`);
+client.on(Events.GuildAuditLogEntryCreate, async auditLog => {
+	// Define our variables.
+	const { action, executorId, target, targetId } = auditLog;
 
-	// Now grab the user object of the person who kicked the member
-	// Also grab the target of this action to double-check things
-	const { executor, target } = kickLog;
+	// Check only for deleted messages.
+	if (action !== AuditLogEvent.MessageDelete) return;
 
-	// Update the output with a bit more information
-	// Also run a check to make sure that the log returned was for the same kicked member
-	if (target.id === member.id) {
-		console.log(`${member.user.tag} left the guild; kicked by ${executor.tag}?`);
+	// Ensure we have the executor cached.
+	const user = await client.users.fetch(executorId);
+
+	if (target) {
+		// We have the message object cached. We can provide a good log here.
+		console.log(`A message by ${target.author.tag} was deleted by ${user.tag}.`);
```

Comment:
```suggestion
		console.log(`A message by ${target.tag} was deleted by ${user.tag}.`);
```

target actually is an user 
https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/GuildAuditLogsEntry.js#L309